tools/verify-release-ancestry.sh

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+#
+# Usage: verify-release-ancestry.sh <commit hash>
+#
+# Exits zero if the provided commit is either an ancestor of main or equal to a
+# hotfix branch (release-branch-*). Exits 1 otherwise.
+#
+set -u
+
+if git merge-base --is-ancestor "$1" origin/main || git name-rev --no-undefined --refs 'refs/remotes/origin/release-branch-*' "$1" ; then
```

Comment:
This doesn't work as expected, unfortunately. For example:
```
$ git name-rev --refs "origin/release-branch-*" --no-undefined 3f9190e63cfd988b8c2267e2068f41812f733387
3f9190e63cfd988b8c2267e2068f41812f733387 origin/release-branch-2024-06-17~2
```
You can see that, rather than say "oh no, I can't find a name for this commit", it found a name just fine: a release branch followed by "~2", meaning "the grandparent of the release branch commit". So this won't actually give a non-zero return code if the commit isn't a release branch tip, it'll only give a non-zero return code if it isn't reachable from a release branch _at all_.

I'm working on a better solution but don't quite have it yet.

Reply:
Works for me, and this was roughly what I thought this was going to look like before I found `name-rev`, which turned out to be trying too hard.